### PR TITLE
Add timeout, increase exception handling and clarify naming

### PR DESCRIPTION
**Pull request by Nindemic**
Signed-off-by: Nindemic <niina.lunden@student.oulu.fi>

### CODE REVIEW

**Comment by JouHouFin** [comment]
Can you fix the missing dollar signs: https://github.com/smaddis/smad-deploy-azure/blob/e7fa9620072102d3d765ff3e6653a023cc40215a/tests/honoscript/send.sh#L46-L53 ?

File: tests/honoscript/send.sh [hunk 1]
```diff
@@ -39,10 +45,14 @@ if [[ "$#" = 2 ]]; then
     case $2 in
         event ) CONTENT='{"alarm": "fire"}';;
         telemetry ) CONTENT='{"temp": 5}';;
```
Comment (msrn) [inline tests/honoscript/send.sh:47]:
```
        event ) CONTENT=echo '{"alarm": "fire"}';;
        telemetry ) CONTENT=echo '{"temp": 5}';;
```

echo string when assinging to variable.

**Review by msrn — CHANGES_REQUESTED** [review]

**Comment by JouHouFin** [comment]
LGTM.

**Comment by msrn** [comment]
LGTM

**Review by JouHouFin — COMMENTED** [review]
Looks good. Feel free to merge.

**Pull request merged**